### PR TITLE
P4 NLP News

### PR DESCRIPTION
**Pull request by leemoispace**
P4 Evaluate a News Article with Natural Language Processing

rubrics:https://learn-udacity.top/fewd586261/Front%20End%20Web%20Developer%20v5.0.0/Part%2005-Module%2001-Lesson%2005_Evaluate%20a%20News%20Article%20with%20Natural%20Language%20Processing/Project%20Rubric%20-%20Evaluate%20a%20News%20Article%20with%20Natural%20Language%20Processing.html

video:

https://user-images.githubusercontent.com/2882655/181092847-e547fa2c-33f7-443b-bf9b-c594dbc842c4.mov

.

### CODE REVIEW

File: p4-evaluate-news-nlp/src/client/js/formHandler.js [hunk 1]
```diff
@@ -1,16 +1,42 @@
-function handleSubmit(event) {
-  event.preventDefault();
+import { checkForName } from "../js/nameChecker";
+let formText;
```
Comment (Martian2Lee) [inline p4-evaluate-news-nlp/src/client/js/formHandler.js]: why initialize here?

**Review by Martian2Lee — CHANGES_REQUESTED** [review]

File: p4-evaluate-news-nlp/src/client/js/formHandler.js [hunk 2]
```diff
@@ -30,7 +43,10 @@ function handleSubmit() {
 
   if (urlCheck) {
     try {
-      getApiKey();
+      getApiKey().then((apiKey) => {
+        return getApiCall(apiKey.api);
+      });
+      //.then((data) => {});
```
Comment (Martian2Lee) [inline p4-evaluate-news-nlp/src/client/js/formHandler.js]: don't include debugging comments in commits

**Review by Martian2Lee — CHANGES_REQUESTED** [review]

File: p4-evaluate-news-nlp/src/client/js/formHandler.js [hunk 3]
```diff
@@ -1,6 +1,15 @@
...
 let formText;
 
+//UI elements
+let resultsId = document.getElementById("results");
+let modelId = document.getElementById("model");
+let agreementId = document.getElementById("agreement");
+let subjectivityId = document.getElementById("subjectivity");
+let confidenceId = document.getElementById("confidence");
+let ironyId = document.getElementById("irony");
+let scoreTagId = document.getElementById("score_tag");
```
Comment (Martian2Lee) [inline p4-evaluate-news-nlp/src/client/js/formHandler.js]: const please

**Review by Martian2Lee — CHANGES_REQUESTED** [review]

File: p4-evaluate-news-nlp/src/client/js/formHandler.js (hunk 1 above)
Comment (leemoispace) [reply p4-evaluate-news-nlp/src/client/js/formHandler.js]: done in [fa51653](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/8/commits/fa516536cb6f3f2079108daa9df3466a2e84a077)

File: p4-evaluate-news-nlp/src/client/js/formHandler.js (hunk 2 above)
Comment (leemoispace) [reply p4-evaluate-news-nlp/src/client/js/formHandler.js]: done in [fa51653](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/8/commits/fa516536cb6f3f2079108daa9df3466a2e84a077)

File: p4-evaluate-news-nlp/src/client/js/formHandler.js (hunk 3 above)
Comment (leemoispace) [reply p4-evaluate-news-nlp/src/client/js/formHandler.js]: done in [5d4ff75](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/8/commits/5d4ff754a3d02f11528cff6bcdb03dd4a1646666)

File: p4-evaluate-news-nlp/src/client/js/formHandler.js (hunk 2 above)
Comment (Martian2Lee) [reply p4-evaluate-news-nlp/src/client/js/formHandler.js]: @leemoispace wrong link 😄

**Review by Martian2Lee — APPROVED** [review]
🚀

**Pull request merged**